lib/datadog/ci/ext/environment.rb

```diff
@@ -35,7 +35,8 @@ module Environment
         module_function
 
         def tags(env)
-          if (_, extractor = PROVIDERS.find { |provider_env_var, _| env.key?(provider_env_var) })
+          _, extractor = PROVIDERS.find { |provider_env_var, _| env.key?(provider_env_var) }
+          if extractor
             tags = send(extractor, env)
```

Comment:
Are we missing tests for this behavior? 🤔

Reply:
Not sure I follow, as all tests will hit this line. Are you talking about a specific corner case?

Reply:
Ha, I think you are referring to how this behaves in older Rubies.
Yeah, it's a weird multi assignment inside a conditional that works on newer Rubies, but it's hard to read, so I extracted it, and it also doesn't work in Ruby 2.1.

Just better this way.